src/v/utils/unresolved_address.h

```diff
@@ -48,6 +41,8 @@ class unresolved_address {
     uint16_t _port{0};
 };
 
+ss::future<ss::socket_address> dns_resolve(unresolved_address);
```

Comment:
I prefer `resolve_dns`, but that's my fault for editing my comment after you looked at it.